### ISSUE TITLE
Gizmo handle for moving object will disappear when I add a new node into the scene on Mac 

### ISSUE THREAD

**Issue by yukoichiro0317** [godotengine/godot#88691]
### Tested versions

v4.2.1.stable.official [b09f793f5]

### System information

MacBook Pro (Retina, 15-inch, Mid 2014) OS : Big Sur ver 11.7.10 Processor 2.5 GHz Quad-Core Intel Core i7 Memory 16 GB 1600 MHz DDR3 Graphics NVIDIA GeForce GT 750M 2 GB Intel Iris Pro 1536 MB

### Issue description

Gizmo handle to move object will disappear when I add a new node into the scene or even when I see the different window, and go back to the scene then the handle disappear.
What I mean “the handle disappear” is normally there is the handle XYZ of course, but when It’s gone, It become only like XY. (Z axis handle gone).
Btw, other handle like Rotation and Size are fine. Only in Move mode. #
<img width="1440" alt="Screen Shot 2024-02-23 at 10 15 55 AM" src="https://github.com/godotengine/godot/assets/52026697/85ec16d0-1d93-480a-8c8e-1880f4bd1592">
↑ in this example, the handle of Y & Z axis are gone.

### Steps to reproduce

Basically, Add some new node into the scene, but it seems randomly.

### Minimal reproduction project (MRP)

N/A
